spring-core/src/main/java/com/baeldung/factorybean/InitializationToolFactory.java

```diff
@@ -0,0 +1,68 @@
+package com.baeldung.factorybean;
+
+import org.springframework.beans.factory.FactoryBean;
+import org.springframework.beans.factory.InitializingBean;
+
+public class InitializationToolFactory implements FactoryBean<Tool>, InitializingBean {
+    private int factoryId;// standard setters and getters
+    private int toolId;// standard setters and getters
+    private String toolName;// standard setters and getters
+    private double toolPrice;// standard setters and getters
+
+    @Override
+    public void afterPropertiesSet() throws Exception {
+        if (toolName == null || toolName.equals("")) {
+            throw new IllegalArgumentException("tool name cannot be empty");
+        }
+        if (toolPrice < 0) {
```

Comment:
It would be better to use some Guava Preconditions or alternative from apache commons than doing it like this